src/tqec/plaquette/rpng_description.py

```diff
@@ -0,0 +1,262 @@
+    
+    The init time is assumed to be 0.
+
+    Attributes:
+        r   ancillaqubit reset basis  
+        g   ancilla qubit measure basis
+        n   time step of ancilla measurement
+    """
+    r: BasisEnum = BasisEnum.X
+    g: BasisEnum = BasisEnum.X
+    n: int = 6
+
+    @classmethod
+    def from_string(cls, rgn_string: str) -> 'RGN':
+        """Initialize the RGN object from a 3-character string"""
+        if len(rgn_string) != 3:
+            raise ValueError('The rgn string must be exactly 3-character long.')
+        r_str, g_str, n_str = tuple(rgn_string)
+
+        try:
+            r = BasisEnum(r_str)
+            g = BasisEnum(g_str)
+            n = int(n_str)
+            return cls(r, g, n)
+        except ValueError:
+            raise ValueError('Invalid rgn string.')
+
+
+@dataclass
+class RPNGDescription:
+    """Organize the description of a plaquette in RPNG format
+    
+    The corners of the square plaquette are listed following the order:
+    tl, tr, bl, br
+    If the ancilla RGN description is not specified, it is assumed 'xx6'
+
+    Attributes:
+        corners RPNG description of the four corners of the plaquette
+        ancilla RGN description of the ancilla
+    """
+    corners: tuple[RPNG, RPNG, RPNG, RPNG]
+    ancilla: RGN = RGN()
+
+    def __post_init__(self):
+        """Validation of the initialization arguments
+        
+        Constraints:
+        - the n values for the corners must be unique
+        - the n values for the corners must be in the interval ]0, ancilla.n[
+        - at least one betwween fields a and p in every corner is z
+        """
+        times = []
+        for rpng in self.corners:
+            if rpng.n:
+                if rpng.n < 1 or rpng.n >= self.ancilla.n:
+                    raise ValueError('The n values must be in ]0, ancilla meas time[.')
+                times.append(rpng.n)
+            # Confirm that at least one betwween fields a and p in every corner is z, if 2Q gate is present
+            if rpng.p and rpng.a != BasisEnum.Z and rpng.p != BasisEnum.Z:
+                raise ValueError('At least one between a and p must be "z" in every corner.')
+        if len(times) != len(set(times)):
+            raise ValueError('The n values for the corners must be unique.')
+        elif len(times) not in [0,2,4]:
+            raise ValueError('Each plaquette must have 0, 2, or 4 2Q gates.')
+
+
+    @classmethod
+    def from_string(cls, corners_rpng_string: str) -> 'RPNGDescription':
+        """Initialize the RPNGDescription object from a (16+3)-character string"""
+        rpng_objs = tuple([RPNG.from_string(s) for s in corners_rpng_string.split(' ')])
+        if len(rpng_objs) != 4:
+            raise ValueError('There must be 4 corners in the RPNG description.')
+        return cls(rpng_objs)
+    
+
+    @classmethod
+    def from_extended_string(cls, ancilla_and_corners_rpng_string: str) -> 'RPNGDescription':
```

Comment:
Add 
```py
from __future__ import annotations
```
at the beginning of the imports and then 

```suggestion
    def from_extended_string(cls, ancilla_and_corners_rpng_string: str) -> RPNGDescription:
```

Reply:
done